core/src/main/java/com/klaytn/caver/tx/type/AbstractTxType.java

```diff
@@ -84,14 +83,56 @@ public AbstractTxType(BigInteger nonce, BigInteger gasPrice, BigInteger gasLimit
         this.from = from;
         this.to = to;
         this.value = value;
+        this.senderSignatureDataSet = new HashSet<>();
     }
 
+  @Deprecated
     protected void setSenderSignatureData(KlaySignatureData signatureData) {
-        this.senderSignatureData = signatureData;
+        addSenderSignatureData(signatureData);
     }
 
+    @Deprecated
     public KlaySignatureData getSenderSignatureData() {
-        return senderSignatureData;
+        Iterator<KlaySignatureData> senderSignatureIterator = senderSignatureDataSet.iterator();
+        if (senderSignatureIterator.hasNext()) {
+            return senderSignatureIterator.next();
+        } else {
+            return null;
```

Comment:
How about return empty value of `KlaySignatrueData`. If you return null, NullPointException could be happen sometimes.

Reply:
This method is to keep the user's code implemented through caver-java before applying multisig. After this PR is reflected, `getSenderSignatureDataSet()` should be used not `getSenderSignatureData()`. If `senderSignatureData` was not initialized in the code that called this method before this PR, it would return null as well.  Also, if you return empty value of `KlaySignatrueData`, user may not notice that they are using the wrong method.

Reply:
In previous version, to return `null` could be happen only when a user puts `null` in constructor or uses set method. But in this case, just return null in the method. If you want to notice wrong data, you could throw exception, not null.

Reply:
`Null` could cause `Internal Error` in the application but empty value does not cause error at least. It will just get error message from Klaytn Network.

Reply:
You know something wrong. Existing code can only set the value when `protected void setSenderSignatureData (KlaySignatureData signatureData)' is called without putting `senderSignatureData` in the construct. If this function is not called, `senderSignatureData` is `null`.

Reply:
Throwing an exception should be modified to include exception handling code in existing user code. Again, this change is intended to minimize modifications to existing users, and `public KlaySignatureData getSenderSignatureData()` is a method that should not be used in the future.

Reply:
`RuntimeException` do not need to change existing user code. I understand this method is depreciated and will be removed. I just wanted to inform you the side effect of `null`.

Reply:
Oh Great. I changed to RuntimeException.